tests/c/Makefile

```diff
@@ -1,29 +1,50 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT license.
 
-#===============================================================================
+#=======================================================================================================================
 # Toolchain Configuration
-#===============================================================================
+#=======================================================================================================================
 
 export CC := gcc
 export CFLAGS := -Wall -Wextra -O3 -I $(INCDIR) -std=c99
 
-#===============================================================================
+#=======================================================================================================================
+# Build Artifacts
+#=======================================================================================================================
 
-all: all-syscalls all-udp-push-pop all-tcp-push-pop
+# C source files.
+export SRC_C := $(wildcard *.c)
 
-all-syscalls:
-	$(CC) $(CFLAGS) syscalls.c -o $(BINDIR)/syscalls $(LIBS)
+# Object files.
+export OBJ := $(SRC_C:.c=.o)
 
-all-udp-push-pop:
-	$(CC) $(CFLAGS) udp-push-pop.c -o $(BINDIR)/udp-push-pop $(LIBS)
+# Suffix for executable files.
+export EXEC_SUFFIX := elf
 
-all-tcp-push-pop:
-	$(CC) $(CFLAGS) tcp-push-pop.c -o $(BINDIR)/tcp-push-pop $(LIBS)
+#=======================================================================================================================
 
-clean: clean-syscalls
+# Builds everything.
+all: $(OBJ) syscalls udp-push-pop tcp-push-pop
 
-clean-syscalls:
-	@rm -rf $(BINDIR)/syscalls
-	@rm -rf $(BINDIR)/udp-push-pop
-	@rm -rf $(BINDIR)/tcp-push-pop
+# Builds system call test.
+syscalls: syscalls.o
+	$(CC) $(CFLAGS) $< -o $(BINDIR)/$@.$(EXEC_SUFFIX) $(LIBS)
```

Comment:
This would work:

```
BUILD_CMD = $(CC) $(CFLAGS) $< -o $(BINDIR)/$@.$(EXEC_SUFFIX) $(LIBS)

syscalls: syscalls.o
	$(BUILD_CMD)

udp-push-pop: udp-push-pop.o
	$(BUILD_CMD)

tcp-push-pop: tcp-push-pop.o
	$(BUILD_CMD)
```

Reply:
Oh, I see!

I thought that you were suggesting having a single rule (like a loop statement) that would expand to the three others.

Anyways, I've applied your suggestion.

Thanks.